static/js/components.js

```diff
@@ -0,0 +1,96 @@
+Vue.component(VueQrcode.name, VueQrcode)
+
+Vue.component('satspay-paid', {
+  props: ['charge', 'timer'],
+  template: `
+    <div class="col-sm-10 col-lg-8">
+        <q-icon name="check" style="color: green; font-size: 21.4em" class="fit"></q-icon>
+        <div class="row text-center q-mt-lg">
+        <div class="col text-center">
+            <q-btn outline v-if="charge.completelink" :loading="charge.paid" :percentage="timer" type="a"
+            :href="charge.completelink" :label="charge.completelinktext">{%raw%}<template v-slot:loading>
+                {{charge.completelinktext}}
+            </template>{%endraw%}</q-btn>
+            <p v-if="charge.completelink" class="q-pt-md">Redirecting after 5 seconds</p>
+        </div>
+        </div>
+    </div>
+    `
+})
+
+Vue.component('satspay-show-qr', {
+  props: ['charge', 'tab', 'value', 'href'],
```

Comment:
- no need to pass the full charge data, the amount is sufficient
```suggestion
   props: ['charge-amount', 'tab', 'value', 'href'],
```

Reply:
- this component knows nothing about `tabs`, maybe the input should be named `type`
- `tab` is just an outer wrapper, in different contexts it can be something else (dialog, expandable-item, etc)